src/codegen.cpp

```diff
@@ -2018,12 +2007,18 @@ static void simple_escape_analysis(jl_value_t *expr, bool esc, jl_codectx_t *ctx
 // ---- Get Element Pointer (GEP) instructions within the GC frame ----
 
 // Emit a gc-root slot indicator
-static Value* emit_local_slot(jl_codectx_t *ctx)
+static Value* emit_local_root(jl_codectx_t *ctx, jl_varinfo_t *vi)
 {
     CallInst *newroot = CallInst::Create(prepare_call(gcroot_func), "", /*InsertBefore*/ctx->ptlsStates);
+    if (vi) {
+        vi->memloc->replaceAllUsesWith(newroot);
```

Comment:
When does it happen that memloc already exists and why can't we just return it if it does?